### PR TITLE
Prirotize registration forms over others

### LINKED ISSUE #1354: When creating a new residential structure, the larval dipping form shows up
APK: 6.0 RC4
user: ams1/Amani123

Create a new residential structure, observe how the larval dipping form is shown
see images below.
image
image
https://github.com/OpenSRP/opensrp-client-reveal/issues/1316
https://github.com/OpenSRP/opensrp-client-reveal/issues/1315

### PR DESCRIPTION
**Pull request by faidvd**
Resolved #1354

### CODE REVIEW

File: opensrp-reveal/src/main/java/org/smartregister/reveal/model/TaskDetails.java [hunk 1]
```diff
@@ -385,11 +385,12 @@ public void setGroupedTaskCodes(String groupedTaskCodeStatusString) {
...
 
-    public boolean isNonResidential() {
-        return getFamilyMemberNames() != null && getFamilyMemberNames().equalsIgnoreCase("NR");
+    public boolean isResidential() {
+        return getFamilyMemberNames() != null && !getFamilyMemberNames().equalsIgnoreCase("NR");
     }
 
-    public int getPendingTasksCount(){
+    public int getPendingTasksCount() {
+        if (getTaskCount() == null) return 1;
```
Comment (Rkareko) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/model/TaskDetails.java:393]: Not sure I get the logic here. why not set it to zero?

File: opensrp-reveal/src/main/java/org/smartregister/reveal/model/TaskDetails.java [hunk 2]
```diff
@@ -385,11 +385,12 @@ public void setGroupedTaskCodes(String groupedTaskCodeStatusString) {
         this.groupedTaskCodes = stringBuilder.toString();
     }
 
-    public boolean isNonResidential() {
-        return getFamilyMemberNames() != null && getFamilyMemberNames().equalsIgnoreCase("NR");
+    public boolean isResidential() {
```
Comment (Rkareko) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/model/TaskDetails.java:388]: Add javados to this method

File: opensrp-reveal/src/main/java/org/smartregister/reveal/model/TaskDetails.java (hunk 1 above)
Comment (faidvd) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/model/TaskDetails.java:393]: taskCount is null for non-grouped tasks. in that case there's only 1 task so returning 1 here.

Comment (faidvd) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/model/TaskDetails.java:393]: will add javadocs

**Review by Rkareko — APPROVED** [review]

**Pull request merged**